### PR TITLE
Feat/setpoint generic

### PR DESCRIPTION
**Pull request by RBinsonB**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | |
| Primary OS tested on | Ubuntu 22.04 |
| Robotic platform tested on | DJI M300 HITL |

---
## How to run this PR feature

<details>
    <summary>Run simulation of the DJI M300 HITL</summary>

Make sure to replace the essentials container by the test container `umdlife/umd-copter-essentials-dev:generic_setpoint`.

</details>

<details>
    <summary>Init and takeoff</summary>

In another terminal, exec into the essential container and run the command to init and takeoff

```bash
docker exec -it essentials /bin/bash
source /ros_entrypoint.sh
source devel/setup.bash
rosservice call /robot128/obtain_release_control_authority "enable_obtain: true" 
rosservice call /robot128/set_local_pos_reference "{}" 
rosservice call /robot128/flight_task_control "task: 4
joystick_command: {x: 0.0, y: 0.0, z: 0.0, yaw: 0.0}
velocity_control_time_ms: 0
pos_threshold_in_m: 0.0
yaw_threshold_in_deg: 0.0"
```

</details>

<details>
    <summary>Send body velocity command</summary>
Once the drone is in the air, publish the following message to control it. You can play with the values but the last element of the array is the flag for controlling the drone in body velocity and should not be changed.

```bash
rostopic pub /robot128/dji_osdk_ros/flight_control_setpoint_generic sensor_msgs/Joy "header:
  seq: 0
  stamp:
    secs: 0
    nsecs: 0
  frame_id: ''
axes: [1.0, 0.0, 0.0, 0.0, 75] 
buttons:
- 0" -r 50
```

</details>

## Description of contribution in a few bullet points

* Adds a `flight_control_setpoint_generic` topic for the DJI M300 to offer the same capacity as on the DJI M600 regarding sending commands in body frame. 

## Description of documentation updates required from your changes

---

## Review Check

- [x] There are no conflicts with target branch.

- [x] Description is intelligible and well written.

- [x] Passes the CI tests.

- [x] README.md is updated.

### CODE REVIEW

File: src/dji_osdk_ros/modules/vehicle_wrapper.cpp [hunk 1]
```diff
@@ -1769,6 +1769,28 @@ static T_OsdkOsalHandler osalHandler = {
     vehicle->flightController->joystickAction();
   }
 
+  void VehicleWrapper::genericCtrl(const JoystickCommand &JoystickCommand, uint8_t flag)
+  {
+    uint8_t HORI  = (flag & 0xC0) >> 6;
+    uint8_t VERT  = (flag & 0x30) >> 4;
+    uint8_t YAW   = (flag & 0x08) >> 3;
+    uint8_t FRAME = (flag & 0x06) >> 1;
```
Comment (biancabnd) [inline src/dji_osdk_ros/modules/vehicle_wrapper.cpp:1777]: just for awareness, why are you shifting here the bits with >>6, >>4, etc?

**Review by biancabnd — APPROVED** [review]
Nice job Robin! I don't know why they removed this functionality in the newer versions of the osdk but is just what we need. I added just one comment which is more of a curiosity. 
I could not test it in HITL today, but I have just one other question, have you tried the other modes besides sending the velocity in body frame? For example, giving it a 91 flag I think it executes position control. Is not really needed, but it would be nice to know that it is indeed generic. 

Overall, ready to be merged! :fire:

File: src/dji_osdk_ros/modules/vehicle_wrapper.cpp (hunk 1 above)
Comment (RBinsonB) [reply src/dji_osdk_ros/modules/vehicle_wrapper.cpp:1777]: This is the main difference with the M600. The different logic enums (HorizontalLogic, VerticalLogic and so on...) have their own flag starting from bit 0 for each of them. They therefor need to be extracted and shifted to start at bit 0 before being casted to their respective enum.

Comment (RBinsonB) [reply src/dji_osdk_ros/modules/vehicle_wrapper.cpp:1777]: And for awareness, I indeed did not test it for different control modes.

**Pull request merged**